src/lava/magma/core/model/py/connection.py

```diff
@@ -1053,8 +1048,26 @@ def _update_traces(self) -> None:
             )
         )
 
+    def run_spk(self, s_in) -> None:
+        """
+        Overrides the Connection Model run_spk function to
+        receive and update y2 and y3 traces.
+        """
+        self._record_pre_spike_times(s_in)
+
+        s_in_bap = self.s_in_bap.recv().astype(bool)
+        y2 = self.s_in_y2.recv()
+        y3 = self.s_in_y3.recv()
+
+        self._record_post_spike_times(s_in_bap)
+
+        y_traces = self._y_traces
+        y_traces[1, :] = y2
+        y_traces[2, :] = y3
+        self._set_y_traces(y_traces)
```

Comment:
Would you update the y2 and y3 traces the same way the Float model is updated? Wouldn't it call update_random_traces for bitApproximate model?